src/Microsoft.VisualStudio.Services.Agent/Blob/DedupManifestArtifactClientFactory.cs

```diff
@@ -97,9 +100,19 @@ private DedupManifestArtifactClientFactory()
                         tracer,
                         cancellationToken);
 
+                    IDedupStoreHttpClient client;
                     // this is actually a hidden network call to the location service:
-                     return Task.FromResult(factory.CreateVssHttpClient<IDedupStoreHttpClient, DedupStoreHttpClient>(connection.GetClient<DedupStoreHttpClient>().BaseAddress));
-
+                    if (domainId == WellKnownDomainIds.DefaultDomainId)
+                    {
+                        client = factory.CreateVssHttpClient<IDedupStoreHttpClient, DedupStoreHttpClient>(connection.GetClient<DedupStoreHttpClient>().BaseAddress);
+                    }
+                    else
+                    {
+                        IDomainDedupStoreHttpClient domainClient = factory.CreateVssHttpClient<IDomainDedupStoreHttpClient, DomainDedupStoreHttpClient>(connection.GetClient<DomainDedupStoreHttpClient>().BaseAddress);
+                        client = new DomainHttpClientWrapper(domainId, domainClient);
+                    }
```

Comment:
can be simplified a bit 
client = (domainId == WellKnownDomainIds.DefaultDomainId) ? line 107 : line 112;

Reply:
It's cleaner this way IMO.